src/scrubber.ts

```diff
@@ -119,4 +133,28 @@ export class Scrubber {
       }
     })
   }
+
+  private splitFields(cfg: ScrubberConfig): StringMap<string[]> {
+    const output: StringMap<string[]> = {}
+    for (const field of Object.keys(cfg.fields)) {
+      const splitField = field.split('.')
+
+      if (splitField.length > 1) {
+        const key = splitField.pop() as string
+        output[key] = splitField
+      }
+    }
+    return output
+  }
+
+  private arraysEqual(a: any[] | undefined, b: any[] | undefined) {
+    if (a === b) return true
+    if (a == null || b == null) return false
+    if (a.length !== b.length) return false
+
+    for (const [i, element] of a.entries()) {
+      if (element !== b[i]) return false
```

Comment:
It works, but in this line it only supports `primitive` comparisons. E.g it'll work to compare numbers/strings, but not nested arrays/objects/maps. If that's ok for your requirements - good.

But also, "simple solution" is to use `_deepEquals` from `js-lib`, as its name says - it can efficiently do deepEqual comparisons (based on a library). Can be used as throw-and-forget solution.

Reply:
I suspected there was something similar around! I realized there was a small bug so I refactored this function to `arrayContainsInOrder` but I'm now also using the _deepEquals for the last part of it!